tests/setup.js

```diff
@@ -1,16 +1,3 @@
-import chai from 'chai';
-import chaiAsPromised from 'chai-as-promised';
-import { config } from 'dotenv';
-
-// Load environment variables from `.env`
-config();
-
-// Load chai as promised for async tests
-chai.use(chaiAsPromised);
-
-// Force chai to show the complete object diff
-chai.config.truncateThreshold = 0;
-
 if (process.env.USE_SERVERLESS_OFFLINE) {
```

Comment:
Now that we've removed dotenv from this file though, can we be sure the env is loaded before it executes? Is it safe to rely on the order of the `setupFiles` array?
Or would this code be better in `setupFilesAfterEnv`? (Assuming it doesn't need to happen before that point...)

Reply:
since it is an array, it respects order - At least that's my experience. To test it, I added a check:

```
if (!process.env.MOCK) {
    throw new Error('bad');
}
```

Run with `MOCK=1` in `.env` executes the tests; without `MOCK` in `.env` it throws the error. I think it's fine to keep it like this

Reply:
Seems quite flakey to me. I think the initial point I made was the wrong way round. Maybe it would be better to remove `dotenv/config` from the `setupFiles` array, and re-add it here. What do you think?

Reply:
Why is it flakey? It performs in the correct order 🙂

Reply:
- Because someone might not realise that the `setup.js` depends on the order of an array in a different file in order to execute properly
- Someone editing `jest.config.js` might not realise that it's crucial to maintain the order of that array
- If the order of that array does changed for some reason, `setup.js` would fail silently, seeing as it's just checking for the truthyness of an env var

Reply:
I don't mind either way, I will change it back to have the env variables setup in the file itself

Reply:
90c0e1d89ea0dcda22ea2d60e4ba3caab1c65050